### PR TITLE
ovl_Bg_Mori_Hineri OK

### PR DESCRIPTION
**Pull request by shawlucas**
its OK

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c [hunk 1]
```diff
@@ -7,8 +13,17 @@
...
+void func_808A3F58(Actor* thisx, GlobalContext* globalCtx);
+
+void func_808A39FC(BgMoriHineri* this, GlobalContext* globalCtx);
+void func_808A3E54(BgMoriHineri* this, GlobalContext* globalCtx);
+void func_808A3C8C(BgMoriHineri* this, GlobalContext* globalCtx);
+void func_808A3BFC(BgMoriHineri* this, GlobalContext* globalCtx);
+void func_808A3BF0(BgMoriHineri* this, GlobalContext* globalCtx);
+void func_808A3D58(BgMoriHineri* this, GlobalContext* globalCtx);
+
+static s16 D_808A43E0 = 0xFFFF;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c:25]:
```suggestion
static s16 D_808A43E0 = -1;
```

File: src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.h [hunk 2]
```diff
@@ -6,9 +6,15 @@
...
 struct BgMoriHineri;
 
+typedef void (*BgMoriHineriActionFunc)(struct BgMoriHineri*, GlobalContext*);
+
 typedef struct BgMoriHineri {
-    /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x20];
+    /* 0x0000 */ DynaPolyActor dyna;
+    /* 0x0164 */ BgMoriHineriActionFunc actionFunc;
+    /* 0x0168 */ s8 objBankIndex;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.h:14]:
```suggestion
    /* 0x0168 */ s8 moriHineriObjIdx;
```
And corresponding temps should also be updated with a more specific name imo.

File: src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c [hunk 3]
```diff
@@ -20,23 +35,251 @@ const ActorInit Bg_Mori_Hineri_InitVars = {
...
+    gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_mori_hineri.c", 618),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(gfxCtx->polyOpa.p++, sDLists[this->dyna.actor.params]);
+    if (this->boxObjIdx > 0) {
+        Matrix_Get(&mtx);
+    }
+    if ((this->actionFunc == func_808A3C8C) && (this->dyna.actor.shape.rot.z != 0)) {
+        func_80093D18(globalCtx->state.gfxCtx);
+        if (this->dyna.actor.params == 0) {
+            Matrix_Translate(-1761.0f, 1278.0f, -1821.0f, 0);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c:251]: MatrixMode enum values should be used when calling Matrix_ functions with a mode argument, like this one.
```suggestion
            Matrix_Translate(-1761.0f, 1278.0f, -1821.0f, MTXMODE_NEW);
```

File: src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c [hunk 4]
```diff
@@ -20,23 +35,251 @@ const ActorInit Bg_Mori_Hineri_InitVars = {
...
+            D_808A43E0 = 0;
+            func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+        } else {
+            this->dyna.actor.draw = NULL;
+            this->actionFunc = func_808A3D58;
+            D_808A43E0 = -1;
+        }
+    }
+    if ((D_808A43E0 > 0) &&
+        (((ACTIVE_CAM->eye.z - this->dyna.actor.posRot.pos.z) < 1100.0f))) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c:220]:
```suggestion
        ((ACTIVE_CAM->eye.z - this->dyna.actor.posRot.pos.z) < 1100.0f)) {
```

File: src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c [hunk 5]
```diff
@@ -20,23 +35,251 @@ const ActorInit Bg_Mori_Hineri_InitVars = {
...
+}
+
+void func_808A3D58(BgMoriHineri* this, GlobalContext* globalCtx) {
+    s16 unk_14E;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hineri/func_808A39FC.s")
+    if ((Flags_GetSwitch(globalCtx, this->switchFlag) &&
+         (this->dyna.actor.params == 0 || this->dyna.actor.params == 2)) ||
+        (!Flags_GetSwitch(globalCtx, this->switchFlag) &&
+         (this->dyna.actor.params == 1 || (this->dyna.actor.params == 3)))) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c:189]:
```suggestion
         (this->dyna.actor.params == 1 || this->dyna.actor.params == 3))) {
```

File: src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c [hunk 6]
```diff
@@ -20,23 +35,251 @@ const ActorInit Bg_Mori_Hineri_InitVars = {
...
+extern UNK_TYPE D_060043D0;
+extern UNK_TYPE D_06006078;
+extern Gfx D_04049FE0[]; // display list for studded dungeon door
+extern Gfx D_06000AE8[];
+extern Gfx D_06001678[];
+
+void BgMoriHineri_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgMoriHineri* this = THIS;
+    s8 objBankIndex;
+    u32 paramTemp;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c:61]:
```suggestion
    u32 switchFlagParam;
```

File: src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c [hunk 7]
```diff
@@ -7,8 +13,17 @@
 void BgMoriHineri_Init(Actor* thisx, GlobalContext* globalCtx);
 void BgMoriHineri_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void BgMoriHineri_Update(Actor* thisx, GlobalContext* globalCtx);
+void func_808A3F58(Actor* thisx, GlobalContext* globalCtx);
```
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c:16]: imo, a line break between these two to make ti clear that its seperate from the main funcs that get set in the init vars
Also, I looked in the game to see what this is and it draws the whole hallway and the room (except for the door and the chute)
So BgMoriHineri_DrawHallAndRoom is probably ok

Comment (Roman971) [reply src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c:16]: I think keeping them together is much better. There is no difference between them in terms of their content/purpose.

Comment (fig02) [reply src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c:16]: I like seeing from a glance which were "additional" from the usual 4, but I do understand that they serve the same function and shouldn't be separated for that reason. disregard my comment lucas

**Pull request merged**